docs/src/proposals/append-vec-storage.md

```diff
@@ -0,0 +1,231 @@
+and it's also more compression friendly for account data.
+
+Specifically:
+
+* Rent-exempt account w/o shared owner uses 80 bytes each.
+* Rent-exempt account w/ shared owner uses 40 + 32 / N bytes each,
+  where N is the number of accounts sharing its owner.
+* Non-rent exempt account uses an extra 8 bytes for rent epoch.
+
+The new AppendVec file is organized into 6 different blocks: header,
+account metas block, account pubkeys block, owners block, account data block,
+and footer.
+
+```
++---------------------------------------------------+
+| header                | 40 bytes                  |
++---------------------------------------------------+
+| account metas block   | 16 bytes per account      |
++---------------------------------------------------+
+| account pubkeys block | 32 bytes per account      |
++---------------------------------------------------+
+| owners block          | 32 bytes per unique owner |
++---------------------------------------------------+
+| account data block    | N + optional 8 bytes      |
+|                       | for rent epoch            |
++---------------------------------------------------+
+| footer                | 40 bytes                  |
++---------------------------------------------------+
+```
+
+### Better Compression for Account Data
+All account-data is put together into one block to make them easier
+to compress, compared to having them stored interleaving with the pubkeys.
+
+### Per-Account Hash is replaced by AppendVec Block Hash
+The per-account hash is removed.  Instead, each block in the new AppendVec file
+has one 32-bytes hash, and the hash for the entire AppendVec is also included
+in the footer.
+
+### Format Details
+
+#### Header
+
+```
++-----------------------------------------+
+| Header                                  |
++-----------------------------------------+
+| format version      | 8 bytes           |
+| account meta offset | 8 bytes           |
+| pubkey_offset       | 8 bytes           |
+| owners_offset       | 8 bytes           |
+| data block offset   | 8 bytes           |
++-----------------------------------------+
+```
+
+#### Account Meta Block
+
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| account meta 1                       |
++--------------------------------------+
+| account meta 2                       |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| account meta N                       |
++--------------------------------------+
+
+```
++-------------------------------------------------------------+
+| Account Meta (16 bytes each)                                |
++-------------------------------------------------------------+
+| owner local id  | 4 bytes                                   |
+|                 | up to 4,294,967,295 owners per append vec |
```

Comment:
currently, max file size is 16G:
https://github.com/solana-labs/solana/blob/37507a2de636efec8bec0b38e794e22230c8d183/runtime/src/append_vec.rs#L42
That could inform our max ranges.

Reply:
> currently, max file size is 16G:

Good to know this.

The local ID represents the ith entry in the owner block.  If the max file size is 16G, then the number of bytes used for the owner's local id can be smaller. (basically, the maximum number of owners allowed in one AppendVec.)